packages/core-data/src/queried-data/reducer.js

```diff
@@ -89,6 +89,19 @@ function items( state = {}, action ) {
 	return state;
 }
 
+const removeQueryItems = () => ( reducer ) => ( state, action ) => {
+	const reconciledState = { ...state };
+	if ( action.type === 'REMOVE_ITEMS' ) {
+		action.query.forEach( ( queryId ) => {
+			remove( reconciledState[ '' ], ( stateQueryId ) => {
```

Comment:
The state at this point contains all the queries, keyed by `stableKey`. Here you are just working with the empty query. This logic needs to work for every query and should probably be in the sub-keyed reducer.

Reply:
I [moved this logic](https://github.com/WordPress/gutenberg/pull/21557/files#diff-f7aec7a322c8fd5825dcb818fc148a9cR33) into the `onSubKey` reducer, got a bit more complex. Will take it out of this PR.

Reply:
`onSubKey` is a generic utility. This doesn't belong there.

>should probably be in the sub-keyed reducer.

That's the reducer that gets passed to the function returned by `flowRight`.